File: back/app/schemas/metrics.py
"""DTOs do dashboard — substituem o `mock-data.ts` do front.

Os nomes dos campos são em português porque é assim que o front já os consome
nos gráficos (dia/pedidos/total, hora/receita...).
"""

from __future__ import annotations

from datetime import date
from enum import StrEnum

from pydantic import BaseModel, Field

from app.schemas.common import Amount, Money


class MetricsRange(StrEnum):
    HOJE = "hoje"
    SEMANA = "semana"
    MES = "mes"


class MetricsSummary(BaseModel):
    total_vendas: Money
    total_pedidos: int
    ticket_medio: Money
    #: Variação % contra o período anterior de mesmo tamanho (pode ser negativa).
    variacao_percentual: Amount
    por_status: dict[str, int] = Field(default_factory=dict)


class DailySales(BaseModel):
    dia: date
    pedidos: int
    total: Money


class ProductSales(BaseModel):
    produto: str
    unidades: int
    receita: Money


class HourlySales(BaseModel):
    hora: int
    pedidos: int
    receita: Money


__all__ = [
    "DailySales",
    "HourlySales",
    "MetricsRange",
    "MetricsSummary",
    "ProductSales",
]

File: front/src/types/metrics.ts
/**
 * Tipos das métricas do dashboard — substituem o antigo `mock-data.ts`.
 * Os nomes dos campos vêm em português porque espelham as views SQL.
 */

/** Períodos aceitos por `GET /api/metrics/summary?range=`. */
export type MetricsRange = 'hoje' | 'semana' | 'mes';

export interface MetricsSummary {
  total_vendas: number;
  total_pedidos: number;
  ticket_medio: number;
  variacao_percentual: number;
  por_status: Partial<Record<string, number>>;
}

/** `GET /api/metrics/daily-sales` */
export interface DailySales {
  dia: string;
  pedidos: number;
  total: number;
}

/** `GET /api/metrics/product-sales` */
export interface ProductSales {
  produto: string;
  unidades: number;
  receita: number;
}

/** `GET /api/metrics/hourly-sales` */
export interface HourlySales {
  hora: number;
  pedidos: number;
  receita: number;
}

// --- Formatos prontos para o Recharts ---------------------------------------

export interface SalesPoint {
  label: string;
  total: number;
  pedidos: number;
}

export interface ProductPoint {
  name: string;
  unidades: number;
  receita: number;
}

export interface HourlyPoint {
  label: string;
  pedidos: number;
  receita: number;
}
